packages/aeproject-cli/aeproject-fire-editor/fire-editor.js

```diff
@@ -0,0 +1,162 @@
+/*
+ * ISC License (ISC)
+ * Copyright (c) 2018 aeternity developers
+ *
+ *  Permission to use, copy, modify, and/or distribute this software for any
+ *  purpose with or without fee is hereby granted, provided that the above
+ *  copyright notice and this permission notice appear in all copies.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
+ *  REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
+ *  AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
+ *  INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
+ *  LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
+ *  OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
+ *  PERFORMANCE OF THIS SOFTWARE.
+ */
+
+const util = require('util');
+const pureExec = require('child_process').exec;
+const exec = util.promisify(pureExec);
+const path = require('path');
+const open = require('open');
+
+const constants = require('./constants.json');
+const infoMessages = constants.MESSAGES;
+
+const moduleName = constants.MODULE_NAME;
+
+const isWindowsPlatform = process.platform === 'win32';
+
+const run = async (options) => {
+
+    try {
+        let cwd = process.cwd();
+        let npmGlobalPath = await getGlobalNpmRoot();
+        npmGlobalPath = npmGlobalPath.stdout.replace('\n', '').trim();
+        let modulePath = path.join(npmGlobalPath, moduleName);
+
+        // check if repo is installed
+        const isInstalled = await isFireEditorInstalled(moduleName);
+        if (!isInstalled) {
+            // install the repo
+            console.log(`====== ${ infoMessages.START_INSTALLING } ======`);
+            await installFireEditorRepo();
+            process.chdir(modulePath);
+            await installModuleDependencies();
+        }
+
+        // update Fire Editor Aepp
+        if (options.update) {
+            // TODO: should we uninstall current version (maybe some settings or files would be deleted like contracts, accounts, etc.)
+            // or only we should update dependencies (npm i)
```

Comment:
maybe it's better to keep the package and only update dependencies.
Otherwise, as you stated we might delete user contracts and I don't think this would be expected by the user.